includes/core/server/request/class-update-intention.php

```diff
@@ -0,0 +1,186 @@
+<?php
+/**
+ * Class file for WCPay\Core\Server\Request\Update_Intent.
+ *
+ * @package WooCommerce Payments
+ */
+
+namespace WCPay\Core\Server\Request;
+
+use WC_Payments;
+use WC_Payments_Http_Interface;
+use WCPay\Core\Exceptions\Server\Request\Invalid_Request_Parameter_Exception;
+use WCPay\Core\Server\Request;
+use WC_Payments_API_Client;
+use WCPay\Payment_Methods\CC_Payment_Method;
+use WCPay\Payment_Methods\Link_Payment_Method;
+
+/**
+ * Request class for creating intents.
+ */
+class Update_Intention extends Request {
+	use Intention;
+	use Level3;
+
+	const IMMUTABLE_PARAMS = [ 'amount' ];
+	const REQUIRED_PARAMS  = []; // This use to be amount and currency, but since it is not required on server API, I will leave it blank.
+	const DEFAULT_PARAMS   = [
+		'receipt_email' => '',
+		'metadata'      => [],
+	];
+
+
+
+	/**
+	 * Sets the intent ID, which will be used in the request URL.
+	 *
+	 * @param string $id Sets the intent ID, which will be used in the request URL.
+	 *
+	 * @throws Invalid_Request_Parameter_Exception
+	 */
+	protected function set_id( string $id ) {
+		$this->validate_stripe_id( $id );
+		$this->id = $id;
+	}
+
+
```

Comment:
There are a lot of cases of unnecessary empty lines in the PR, please go through them ana minimze them (1 in this case).